File: projects/spring-boot-restv2/demo/src/main/java/com/sourabh/demo/iservice/ICustomerService.java
package com.sourabh.demo.iservice;

import com.sourabh.demo.model.Customer;
import org.bson.Document;

import java.util.List;

public interface ICustomerService {
    public Customer addCustomer(Customer customer);
    public List<Customer> getAll();
    public Document getOccupantsCounts();
    public Document getOccupantsCountsHistory();
    public Document getDateWiseBooking();
    public Customer getCustomer(int id);
    public float calculateFare(int days, float baseFare);
}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/iservice/ICustomerService.kt
package com.sourabh.hotelreservation.iservice

import com.sourabh.hotelreservation.model.Customer
import com.sourabh.hotelreservation.repository.TypeOccupant
import org.bson.Document
import org.springframework.data.mongodb.core.aggregation.AggregationResults

interface ICustomerService {

    fun addCustomer(customer:Customer):Customer
    fun getAll():MutableList<Customer>
    fun getOccupantsCounts(): Document
    fun getOccupantsCountHistory(): Document
    fun getCustomer(id:Int):Customer
    fun calculateFare(days:Int, baseFare:Float):Float
    fun getDateWiseBookings():Document
}